src/box/lua/schema.lua

```diff
@@ -812,18 +812,25 @@ box.schema.space.create = function(name, options)
             box.error(box.error.SPACE_EXISTS, name)
         end
     end
+    local _schema = box.space._schema
     local id = options.id
     if not id then
-        local _schema = box.space._schema
         local max_id = _schema:update({'max_id'}, {{'+', 2, 1}})
-        if max_id == nil then
-            id = _space.index.primary:max().id
+        id = _space.index.primary:max().id
```

Comment:
This won't work if there's a space with the max id:

```
tarantool> box.schema.space.create('test', {id = 2147483647})
---
- engine: memtx
  before_replace: 'function: 0x416a02a0'
  field_count: 0
  is_sync: false
  on_replace: 'function: 0x416a0278'
  temporary: false
  index: []
  is_local: false
  enabled: false
  name: test
  id: 2147483647
- created
...

tarantool> box.schema.space.create('test2')
---
- error: 'Failed to create space ''test2'': space id is too big'
...
```

Reply:
Reworked